core/any_error.rs

```diff
@@ -0,0 +1,55 @@
+use std::any::{Any, TypeId};
+use std::convert::From;
+use std::error::Error;
+use std::ops::Deref;
+
+pub type ErrBox = Box<dyn AnyError>;
+
+pub trait AnyError: Error + Any + Send + Sync + 'static {}
```

Comment:
It would be good to have a comment about the necessity of these various traits. (EG that deno is multithreaded and we need to be able to handle errors across threads)